pkg/internal/infrastructure/terraform.go

```diff
@@ -145,6 +148,19 @@ func ComputeTerraformerTemplateValues(
 		"outputKeys": outputKeys,
 	}
 
+	if config.Networks.PrivateServiceConnect != nil && config.Networks.PrivateServiceConnect.Enabled {
```

Comment:
It would be sufficient but I went for a more explicit enablement. How about this check: if PSC struct it not nil, then both are true:
+ we consider PSC enabled
+ endpointIP **must** not be empty (enforced via validation).